torch/csrc/jit/runtime/register_cuda_ops.cpp

```diff
@@ -67,6 +68,15 @@ RegisterOperators const reg({
         [](Stack* stack) {
           auto v = pop(stack);
           auto s = v.toCustomClass<torch::jit::CUDAStream>();
+          auto stream_device_idx = static_cast<int64_t>(s->device_index());
+          auto cur_device_idx =
+              static_cast<int64_t>(c10::cuda::current_device());
+          // If the stream is not on the current device, change the
+          // device to the device of the stream.
+          if (cur_device_idx != stream_device_idx) {
+            c10::cuda::set_device(
+                static_cast<c10::DeviceIndex>(stream_device_idx));
+          }
```

Comment:
This change seems to be unrelated to the rest of the commit ("Update Namespace from cuda to _cuda"). I know this is needed, but let's land it in a separate commit.